### ISSUE TITLE
Anvil ignores eth_getLogs filters when running in forked mode

### ISSUE THREAD

**Issue by ffoxdd** [foundry-rs/foundry#1827]
### Component

Anvil

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (0342bc2 2022-06-03T00:04:00.330112Z)

### Describe the bug

I'm running an anvil node in rust using anvil::spawn.

When I use rust-web3 to query for logs using the spawned server's web socket url, it ignores my topic filter and instead returns all logs for the given contract.

As a result I get this error:

`"Fork Error: JsonRpcClientError(ProviderError(JsonRpcError(JsonRpcError { code: -32602, message: \"Log response size exceeded. You can make eth_getLogs requests with up to a 2K block range and no limit on the response size, or you can request any block range with a cap of 10K logs in the response. Based on your parameters and the response size limit, this block range should work: [0x0, 0x721655]\"`

It works fine when I use a non-anvil web socket endpoint.

**Comment by mattsse** [comment]
what are the filter parameters for `eth_getLogs`? from: 0 to : 7476821? because that's a lot for HTTP, so the error message makes sense, right?

> It works fine when I use a non-anvil web socket endpoint.

and HTTP?

**Comment by ffoxdd** [comment]
The fact that I'm searching through a lot of blocks is actually how I surfaced this bug: I'm performing a getLogs query with a filter for a specific address value for topic1. When filtering for that specific topic1 value the result set is only ~100 logs. Without it there are many thousands.

I get my ~100 log result set when I use a regular alchemy url, but it fails as soon as I swap out that url for the one for a spawned forked anvil node.

The error message is a little confusing there: it's not saying that I used (0, 7476821) as a filter parameter, but rather that it expects that I could use that more restrictive range if I wanted to get a more manageable result set.

I haven't checked what happens using http transport -- I'll throw together a minimal test case and reply back.
